docs/TROUBLESHOOTING.md

```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
 
 try
 
-```
+```bash
 gcloud alpha billing projects link projects/some-project --billing-account XXXXXX-XXXXXX-XXXXXX
 ```
 
 If output states `Cloud billing quota exceeded`, you can use the [Request Billing Quota Increase](https://support.google.com/code/contact/billing_quota_increase) form to request a billing quota increase.
+
+### Terraform Error acquiring the state lock
+
+**Error message:**
+
+```text
+Error: Error acquiring the state lock
+```
+
+**Cause:**
+
+This message means you are trying to execute a Terraform command with the [Terraform State locked](https://www.terraform.io/language/state/locking).
+
+If the Terraform process was unable to finish due to an unexpected event, i.e build timeout or terraform process killed. It will keep the Terraform State **locked**.
+
+**Solution:**
+
+Keep in mind that after removing the Terraform State lock you will need to [review and manipulate](https://www.terraform.io/cli/state#manipulating-terraform-state) your Terraform State and maybe [import](https://www.terraform.io/cli/import#import) some resources.
+
+Here are some sample commands on how to remove the Terraform State lock.
```

Comment:
I would rephrase this in the context of unlocking a single root config like https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/2-environments/envs/development. So you can guide the user through git clone repo, cd to env, change backend if needed, run tf locally to confirm state is locked, locally unlock state, do a plan to validate no diff or look at import/state rm for massage the state.

Reply:
Our last instruction could also be to review what caused the lock to happen. If this was due to a build time our recommend increasing build timeout https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/build/cloudbuild-tf-apply.yaml#L15

Reply:
done